eth/downloader/downloader.go

```diff
@@ -419,13 +455,11 @@ func (d *Downloader) synchronise(id string, hash common.Hash, td *big.Int, mode
 	d.cancelPeer = id
 	d.cancelLock.Unlock()
 
-	defer d.cancel() // No matter what, we can't leave the cancel channel open
+	defer d.Cancel() // No matter what, we can't leave the cancel channel open
```

Comment:
So as you'll see below the exported `Cancel` waits for pending fetchers to finish, while the private method just kills hard.